IBM-Cloud/ibm-developer-extension-vscode
# feat: Updated to latest cli 2.5.0

# Context
The PR will update the extension to use the latest version of the CLI. In addition, all references to **bluemix** has been rebranded to **IBM Cloud**

# Steps to Test
### Setup
1. Install NPM
2. Install the latest IBM Cloud CLI
3. Install Visual Studio Code Extension Manager CLI (`vsce`)
```
npm install -g vsce
```
4. Install VSCode
```
vsce package
```
5. Change directory to the root of the ibm-developer-extension-vscode project
6. Build extension
7. Open VSCode
8. Click on **Extension** then click on the `...` on the top left side of the panel
9. Click on the Install From VSX in the dropdown
10. Select the `.vsx` file that was built in step 4
11. Verify that the ibm-developer-0.0.15.vsix was created
12. Create a namespace in IBMCloud Container Registry in production
  12.1 You will need this when deploying container
13. Create or use an existing K8 cluster

### Verify rebranding
1. Check sourcecode and README **DOES NOT** reference **bluemix**
  1.1 You should see **bluemix** renamed  to **IBMCloud**

### Test create, running and deploying app development through IKS workflow
1. Open VSCode to a blank page
2. Open up the command palette using the shortcut: *Ctrl+Shift+P*
3. Enter `ibmcloud login --sso` and select command
  3.1 Make sure that you are using production API endpoint
4. Enter `ibmcloud dev create` and select command
5. Verify that terminal window is opened in VSCode
6. Follow and respond to the prompts presented
7. Verify that new project is created.
8. Open up new project in VSCode
9. Open command palette and enter `ibmcloud dev build (release)` and select command
10. Verify that project is built successfully
11. Open command palette and enter `ibmcloud dev run` and select command
12. Verify that you view app by visiting url in browser
13. Set the following properties in the `cli-config,.yml` file
  ```
    deploy-target: "container"
    deploy-image-target: "icr.io/<namespace>/<image-name>"
    ibm-cluster: "<cluster-name>"

  ```
14. Verify that hosted app url is shown in the Output panel
15. Visit hosted app in your browser
